modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/APMAgentSettings.java

```diff
@@ -114,15 +115,130 @@ public void setAgentSetting(String key, String value) {
     private static final String APM_SETTING_PREFIX = "tracing.apm.";
 
     /**
-     * A list of APM agent config keys that should never be configured by the user.
+     * Allow-list of APM agent config keys users are permitted to configure.
+     * @see <a href="https://www.elastic.co/guide/en/apm/agent/java/current/configuration.html">APM Java Agent Configuration</a>
      */
-    private static final List<String> PROHIBITED_AGENT_KEYS = List.of(
-        // ES generates a config file and sets this value
-        "config_file",
-        // ES controls this via `telemetry.metrics.enabled`
-        "recording",
-        // ES controls this via `apm.enabled`
-        "instrument"
+    private static final Set<String> PERMITTED_AGENT_KEYS = Set.of(
+        // Circuit-Breaker:
+        "circuit_breaker_enabled",
+        "stress_monitoring_interval",
+        "stress_monitor_gc_stress_threshold",
+        "stress_monitor_gc_relief_threshold",
+        "stress_monitor_cpu_duration_threshold",
+        "stress_monitor_system_cpu_stress_threshold",
+        "stress_monitor_system_cpu_relief_threshold",
+
+        // Core:
+        // forbid 'enabled', must remain enabled to dynamically enable tracing / metrics
+        // forbid 'recording' / 'instrument', controlled by 'telemetry.metrics.enabled' / 'tracing.apm.enabled'
+        "service_name",
+        "service_node_name",
+        // forbid 'service_version', forced by APMJvmOptions
+        "hostname",
+        "environment",
+        "transaction_sample_rate",
+        "transaction_max_spans",
+        "long_field_max_length",
+        "sanitize_field_names",
+        "enable_instrumentations",
+        "disable_instrumentations",
+        "enable_experimental_instrumentations", // TODO TBD allow disabling this, conflicts with APMJvmOptions
+        "unnest_exceptions",
+        "ignore_exceptions",
+        "capture_body",
+        "capture_headers",
+        "global_labels",
+        "instrument_ancient_bytecode",
+        "context_propagation_only",
+        "classes_excluded_from_instrumentation",
+        "trace_methods",
+        "trace_methods_duration_threshold",
+        "central_config", // TODO TBD disabled in APMJvmOptions defaults, does this impact usage of config_file?
```

Comment:
Not sure about this one, could it impact usage of / conflict with `config_file`?

Reply:
Let's remove this one as it impacts `config_file`.

Reply:
👍